extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -405,7 +417,7 @@ void registerBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans, Combi
 
         additionalBeans.produce(AdditionalBeanBuildItem.builder().setUnremovable()
                 .addBeanClasses(JPAConfig.class, TransactionEntityManagers.class,
-                        RequestScopedEntityManagerHolder.class)
+                        RequestScopedEntityManagerHolder.class, DataSourceTenantConnectionResolver.class)
```

Comment:
Could you avoid registering this bean as "unremovable" when multitenancy is not being used?

Reply:
OK, I changed it.